shacl12-core/index.html

```diff
@@ -2702,6 +2710,27 @@ <h3>Data Graph</h3>
 					several related ontologies, pass all of them to the SHACL processor (together or one by one), do not rely on <code>owl:imports</code> links.
 				</p>
 			</section>
+
+			<section id="subClassOfInShapesGraph">
+				<h3>Graph for rdfs:subClassOf Triples</h3>
+				<p>
+					Some features of SHACL (such as 
+					<a href="#ClassConstraintComponent"></a>, 
+					<a href="#targetClass"></a> and 
+					<a href="#implicit-targetClass"></a>) rely on the notion
+					of <a>SHACL type</a> to determine whether a <a>node</a> is a <a>SHACL instance</a> of a given class.
+					By default, this is determined by looking up <code>rdfs:subClassOf</code> and <code>rdf:type</code> <a>triples</a>
+					in the <a>data graph</a>.
+					However, in some cases this is impractical as <code>rdfs:subClassOf</code> triples are often stored as part
+					of the class/shape definitions and not the instance data.
+				</p>
+				<p>
+					SHACL processors SHOULD offer a parameter <code>subClassOfInShapesGraph</code> that, if set to <code>true</code>
+					should alter the definition of <a>SHACL Type</a> so that the <code>rdfs:subClassOf</code> triples are queried
+					from the <a>shapes graph</a> instead of the <a>data graph</a>.
```

Comment:
```suggestion
					from the <a>shapes graph</a> in addition to the <a>data graph</a>.
```

Why omit consideration of `rdfs:subClassOf` statements in the data graph when this mode is enabled?

Reply:
That COULD work too. I went with the simpler solution because querying a union graph would be more complex, and make optimizations harder. If we know that we only look in the shapes graph, all class closures could be pre-computed into HashSets, instead of having to walk the hierarchy over and over.

Also, what about undesirable cases where some subClassOf triples are in the data graph and others in the shapes graph. That sounds like a mess to me.

Reply:
Are there realistic scenarios where people want to mix and match?

Reply:
> Are there realistic scenarios where people want to mix and match?

I can imagine some cases e.g. having an upper level ontology in your shapes graph and then lower level ontologies in your data graph. I have no data on whether these appear in the wild.

Reply:
I can see use cases coming from defining higher-level category classes that join together classes in the data graph.

Use case:

* Data graph goes as ontologically far as classifying `ex:Cat` and `ex:Dog` and putting them under `ex:Mammal`.
* Shapes graph includes an overarching class `ex2:WarmBloodedAnimal`.

Use case:

* Data graph is a union of credit card receipts and cash receipts.
* Shapes graph includes an overarching class `ex3:FinancialTransaction`.

I think mix-and-match should be supported by default.  I'm late coming to this PR and Issue, and a little curious about the motivation for it being "Opt-in" to look for `rdfs:subClassOf` in the shapes graph.  Just the union-performance matter?

Reply:
I have no opposition to generalizing this flag to be the union of data and shapes graph. There seem to be use cases that people need, so why not, even if it can cause some potential performance penalties.